data_structures/stack_using_queue.cpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * @file
+ * @brief Implement a last in first out (LIFO) stack using queues. The implemented stack should support all the functions of a normal queue (push, top, pop, and empty).
+ * @author [Neha Labhasetwar](https://github.com/nehalabhasetwar)
+ */
+
+#include <iostream>  // for standard IO operations
+#include <cassert>   // defines one macro function (assert) that can be used as a standard debugging tool
+#include <queue>     // This header is part of the containers library, designed to operate in FIFO context
+/**
+ * @namespace data_structures
+ * @brief In this approach we will see how to Implement Stack using Single Queue,
+ * 1.Push the element to the Queue :
+ *   -Check the size of Queue and store it in variable "s".
+ *   -Iterate from 0 to "s" and Pop the element from queue and push the same element to Queue. By doing this, last inserted data will come at first position in Queue.
+ * 2.Pop Operation:
+ *   -Pop the elements from Queue.
+ * 3.top Operation:
+     -Return front element from Queue.
+*/
+namespace data_structures {
+/**
+ * @namespace stack_using_queue
+ * @brief Functions for [Stack using queue](https://leetcode.com/problems/implement-stack-using-queues) implementation
+ */
+namespace stack_using_queue {
+/**
+ * @brief Assuming we already have a class implemented for Queue, we will first design the class for Stack.
+ * It will have the methods push() , pop() , top() , and empty() and one queue.
+ */
+class MyStack
+{
+    /** initialize queue */
+    std::queue <int> q;      //empty queue q of int type
```

Comment:
Consider using `uint64_t` rather than `int` (check other parts of the code).
```suggestion
    std::queue <uint64_t> q{};      // empty queue q of uint64_t type
```